docs: Add Publisher & Developer documentation (#5)
Automattic/newspack-tecnavia: PUBLISHERS.md

# Newspack-Tecnavia Plugin Setup Guide

The Newspack-Tecnavia plugin allows publishers to integrate Tecnavia's e-Edition services into their WordPress site, providing readers with access to digital editions directly from their WordPress website based on authentication configurations.

## Prerequisites
This Plugin has the following plugins as dependencies:-

1. Newspack.
2. Woocommerce.

> 3. Woocommerce Subscriptions.

Reviewer: either Woo Subscriptions or Memberships if you want to lock access to Tecnavia bhind a subscription or a membership

4. Woocommerce Memberships.

> Please install, activate and configure the above plugins before activating and configuing this plugin.

Reviewer: configuring

## Installation
1. **Download the Plugin**:
	 - Obtain the latest version of the plugin from the [GitHub repository](https://github.com/Automattic/newspack-tecnavia).

2. **Upload to WordPress**:
	- Navigate to `Plugins` > `Add New` > `Upload Plugin`.
	- Select the downloaded plugin file and click `Install Now`.

3. **Activate the Plugin**:
   - Click `Activate` to enable the plugin.
	- Check if there is a notice to install the required plugins.

## Configuration
1. **Access Plugin Settings**:
   - In the WordPress admin dashboard, navigate to `Woocommerce` > `Settings` > `Newspack Tecnavia Integration`.

   ![Plugin Settings](https://github.com/user-attachments/assets/6705be15-8f7b-4e2d-b7a3-0a140691186b)

2. **Add Tecnavia Configurations**:
   - **E-edition URL Endpoint**: This will be the endpoint on your site which will lead users to the e-Edition.

>    For example:- an endpoint could be `e-edition`. This would mean that the e-Edition URL would be `https://your-site.com/e-edition`.

Reviewer: Add: You can add this URL to any menu in your site (or really anywhere) as an entry point for readers to access the e-edition. (if the reader is not logged in or doesn't have access to the e-edition based on the rules you set, they will be redirected to the fallback page)

>    - **E-edition Endpoint Link Label**: This will be the text of the link that users will click on to access the e-Edition. This link will be displayed on the users `my-account` page.

Reviewer: ```suggestion
   - **E-edition Endpoint Link Label**: This will be the label of the link that is automatically added to the "My Account" menu in the readers' "My Account" page.
```

   - **Tecnavia URL**: This is the URL of your Tecnavia e-Edition. The user will be redirected to this URL when they click on the e-Edition link.

>    - **Fallback Page**: Select the page where users will be redirected if they do not have access to the e-Edition. You can use this page to funnel users into e-edition. For instance, it contain information on which memberships/subscriptions could get you an e-edition access.

Reviewer: ```suggestion
   - **Fallback Page**: Select the page where users will be redirected if they do not have access to the e-Edition. You can use this page to funnel users into e-edition. For instance, it can contain information about the e-Edition and the CTAs to buy access to it.
```

   ![Integration Settings](https://github.com/user-attachments/assets/2126ecc2-2a4c-4c48-a590-160c9ce4d61e)

3. **Configure User Access**: 
   - The Plugin allows you to restrict access to the e-Edition based on your users role, memberships or subscriptions.
   - Following are the options to configure user's access to the e-Edition:
     - **All registered users**: If this option is selected, all registered users will have access to the e-Edition. Regardless of their role, membership or subscription.

> 	 - **User Role**: Select the user role that will have access to the e-Edition.

Reviewer: roles, plural

> 	 - **Membership**: Select the memberships that will grant access to the e-Edition. If a user has any of the selected memberships, the user will be granted access to the e-Edition.

Reviewer: ```suggestion
	 - **Membership**: Select the membership plans that will grant access to the e-Edition. If a user has any of the selected plans, the user will be granted access to the e-Edition.
```

> 	 - **Subscription**: Select the subscription products that will grant access to the e-Edition. If a user has any of the selected subscription product, the user will be granted access to the e-Edition.

Reviewer: ```suggestion
	 - **Subscription**: Select the subscription products that will grant access to the e-Edition. If a user has any of the selected subscription products, the user will be granted access to the e-Edition.
```
